Kitodo/src/main/java/org/kitodo/production/migration/NewspaperProcessesMigrator.java

```diff
@@ -0,0 +1,745 @@
+            String firstSimpleMetadataValue = MetadataEditor.readSimpleMetadataValues(child, simpleMetadataView).get(0);
+            switch (Integer.signum(firstSimpleMetadataValue.compareTo(value))) {
+                case -1:
+                    index++;
+                    break;
+                case 0:
+                    return child;
+                case 1:
+                    break FOR_LOOP;
+                default:
+                    throw new IllegalStateException("complete switch");
+            }
+        }
+        IncludedStructuralElement computed = new IncludedStructuralElement();
+        MetadataEditor.writeMetadataEntry(computed, simpleMetadataView, value);
+        includedStructuralElement.getChildren().add(index, computed);
+        return computed;
+    }
+
+    /**
+     * Adds a date to get a complete date. In Production versions before 2.2,
+     * the date is stored incompletely (as an integer). This is supplemented to
+     * ISO if found. Otherwise just returns the date.
+     *
+     * @param includedStructuralElement
+     *            the included structural element that contains the date
+     * @param previousLevel
+     *            previous part of date
+     * @return ISO date
+     */
+    private static String getCompletedDate(IncludedStructuralElement includedStructuralElement, String previousLevel) {
+        String date = MetadataEditor.getMetadataValue(includedStructuralElement, FIELD_TITLE_SORT);
+        if (!date.matches("\\d{1,2}")) {
+            return date;
+        }
+        logger.debug("Found integer date value ({}), supplementing to ISO date", date);
+        StringBuilder composedDate = new StringBuilder();
+        composedDate.append(previousLevel);
+        composedDate.append('-');
+        if (date.length() < 2) {
+            composedDate.append('0');
+        }
+        composedDate.append(date);
+        return composedDate.toString();
+    }
+
+    /**
+     * Creates an overall process as a representation of the newspaper as a
+     * whole.
+     *
+     * @throws ProcessGenerationException
+     *             An error occurred while creating the process.
+     * @throws IOException
+     *             An error has occurred in the disk drive.
+     */
+    public void createOverallProcess() throws ProcessGenerationException, IOException, DataException, DAOException {
+        final long begin = System.nanoTime();
+        logger.info("Creating overall process {}...", title);
+
+        ProcessGenerator processGenerator = new ProcessGenerator();
+        processGenerator.generateProcess(templateId, projectId);
+        overallProcess = processGenerator.getGeneratedProcess();
+        overallProcess.setTitle(getTitle());
+        processService.save(overallProcess);
+        ServiceManager.getFileService().createProcessLocation(overallProcess);
+        overallWorkpiece.getRootElement().getMetadata().addAll(overallMetadata);
+        addToBatch(overallProcess);
```

Comment:
why is it added to the batch?

Reply:
The batch represents the entire newspaper. For example, you can use it to export the entire newspaper. Therefore, I think, the additional processes created for this newspaper belong there. Otherwise you would export only the individual issues of the newspaper after the migration, without their hierarchical structures.